plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -229,6 +231,15 @@ public ClickHouseClient(
                         .build());
     }
 
+    private String getClusterInfo()
+    {
+        StringBuilder clusterInfoBuilder = new StringBuilder();
+        if (clusterName.isPresent() && !isNullOrEmpty(clusterName.get())) {
+            clusterInfoBuilder.append(" ON CLUSTER ").append(clusterName.get()).append(" ");
+        }
+        return clusterInfoBuilder.toString();
```

Comment:
I expect the outer spaces are handled in the caller side.
```suggestion
        return clusterName.map("ON CLUSTER %s"::formatted).orElse("");
```